### DISCUSSION TITLE
How to interpret same address failing after rotating 2 modules

### DISCUSSION THREAD

**Discussion by aksh1618** [memtest86plus/memtest86plus#358] [Q&A]
In the [Trouble-shooting Memory Errors](https://memtest.org/readme#trouble-shooting-memory-errors) section, the technique of rotating modules has been suggested, which says that it can only be used if there are three or more modules in the system. Could you please suggest why this is 3 and not 2? For my system which has only 2 modules, after rotating the 2 modules and running memtest86+ again, the same address gets reported as the failing address (`5.39GB`). Can I interpret this result as meaning that the issues is not with the memory modules?

Some background:
- After facing random browser crashes, I finally ran memtest86+ which revealed a lot of issues (7K+) at the address `5.39GB`.
- As a next step I decided to clean my laptop (AMD Ryzen 7 4800H) and rotate the modules (2x8GB)
- Post this the test came out as PASS (3 passes). All crashes also stopped
- 5 days later (today), I again faced a crash, so I ran memtest86+ again, which again logged a lot of failures (5K+) at the exact same address `5.39GB`
- Then I went to the README to figure out what to do next, where I noticed the condition of having 3 or more modules

**Comment by debrouxl** [comment]
On this platform, with 2 memory modules, the memory controller ought to be running in dual-channel mode, and the usual dual-channel mode is interleaved. In such a case, I'd expect the failing address to be either the same, or marginally different (within several bytes), when swapping the modules.
Try running memtest86+ again with a single memory module at a time; the memory issue should show up when the faulty module is plugges in, unless maybe if it's bandwidth-related (single memory stick => single channel = lower total bandwidth).

memtest86+ can print badram patterns to make Linux avoid the faulty memory addresses.
